code/math/vecmat.cpp

```diff
@@ -2509,29 +2509,32 @@ void vm_matrix_sub2(matrix* dest, const matrix* src)
 */
 bool vm_inverse_matrix(matrix* dest, const matrix* m)
 {
-	matrix inv;	// create a temp matrix so we can avoid getting a determinant that is 0
+	// Use doubles here because this is used for ship inv_mois and we could be dealing with extremely small numbers
+	double inv[3][3];	// create a temp matrix so we can avoid getting a determinant that is 0
 
 	// Use a2d so it's easier for people to read
-	inv.a2d[0][0] = -m->a2d[1][2] * m->a2d[2][1] + m->a2d[1][1] * m->a2d[2][2];
-	inv.a2d[0][1] = m->a2d[0][2] * m->a2d[2][1] - m->a2d[0][1] * m->a2d[2][2];
-	inv.a2d[0][2] = -m->a2d[0][2] * m->a2d[1][1] + m->a2d[0][1] * m->a2d[1][2];
-	inv.a2d[1][0] = m->a2d[1][2] * m->a2d[2][0] - m->a2d[1][0] * m->a2d[2][2];
-	inv.a2d[1][1] = -m->a2d[0][2] * m->a2d[2][0] + m->a2d[0][0] * m->a2d[2][2];
-	inv.a2d[1][2] = m->a2d[0][2] * m->a2d[1][0] - m->a2d[0][0] * m->a2d[1][2];
-	inv.a2d[2][0] = -m->a2d[1][1] * m->a2d[2][0] + m->a2d[1][0] * m->a2d[2][1];
-	inv.a2d[2][1] = m->a2d[0][1] * m->a2d[2][0] - m->a2d[0][0] * m->a2d[2][1];
-	inv.a2d[2][2] = -m->a2d[0][1] * m->a2d[1][0] + m->a2d[0][0] * m->a2d[1][1];
-
-	float det = m->a2d[0][0] * inv.a2d[0][0] + m->a2d[0][1] * inv.a2d[1][0] + m->a2d[0][2] * inv.a2d[2][0];
+	inv[0][0] = -(double)m->a2d[1][2] * m->a2d[2][1] + (double)m->a2d[1][1] * m->a2d[2][2];
```

Comment:
I'd recommend putting `double` on all four operands here.  The compiler probably promotes everything to double before doing the math, but putting double everywhere makes it explicit for people reading the code.